src/Symfony/Component/Translation/CatalogueMetadataAwareInterface.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation;
+
+/**
+ * CatalogueMetadataAwareInterface.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+interface CatalogueMetadataAwareInterface
+{
+    /**
+     * Gets catalogue metadata for the given domain and key.
+     *
+     * Passing an empty domain will return an array with all catalogue metadata indexed by
+     * domain and then by key. Passing an empty key will return an array with all
+     * catalogue metadata for the given domain.
+     *
+     * @return mixed The value that was set or an array with the domains/keys or null
+     */
+    public function getCatalogueMetadata(string $key = '', string $domain = 'messages'): mixed;
+
+    /**
+     * Adds catalogue metadata to a message domain.
+     */
+    public function setCatalogueMetadata(string $key, mixed $value, string $domain = 'messages');
+
+    /**
+     * Deletes catalogue metadata for the given key and domain.
+     *
+     * Passing an empty domain will delete all catalogue metadata. Passing an empty key will
```

Comment:
Is there a use case for deleting all metadata? Maybe that should not be possible to prevent possible problems when a provider did send an empty meta-key or missing domain.

Reply:
I don't think so, tbh I've literally copy/pasted the MetadataAwareInterface and adapted it for catalogue's metadata.

Reply:
why do we need CatalogueMetadataAwareInterface then? can't we use MetadataAwareInterface instead?

Reply:
AFAIK, the `MetadataAwareInterface` is related to translation keys and not the catalogue itself.

Reply:
Looking at these interfaces, it's not obvious how they are different and even what they are about. 
Could you add a description on the interface? Or on `MetadataAwareInterface`, or both maybe?